scripts/xMetric/addTransactor.ts

```diff
@@ -0,0 +1,26 @@
+import { ethers } from "hardhat";
+import fs from "fs";
+
+async function main() {
+  const newTrasactor = "";
+
+  const network = process.env.HARDHAT_NETWORK;
+  console.log("adding transactor on: ", network);
+
+  const data = await fs.readFileSync("./deployments/" + network + "/Xmetric.json", "utf8");
+  const object = JSON.parse(data);
+  const address = object.address;
+
+  const MyContract = await ethers.getContractFactory("Xmetric");
```

Comment:
Guess I'm confused from the hardhat documentation

> The Contract s and ContractFactory  returned by these helpers are connected to the first [signer](https://docs.ethers.io/v5/single-page/#/v5/api/signer/) returned by getSigners by default.

How does it know which signer it is - I'm assuming the owner of the contract should be the signer but where is the relationship established?

Reply:
it pulls from `hardhat.config.ts` - like on ropsten it would use this account: https://github.com/MetricsDAO/core-evm-contracts/blob/main/hardhat.config.ts#L32

Reply:
I don't see anything for polygon on hardhat config do you want to add and just use your .env file?

Reply:
yeah I was gonna do it live haha but good point we can prep.